api/multicluster/v1alpha1/cluster.proto

```diff
@@ -14,17 +14,43 @@ import "skv2/api/core/v1/core.proto";
     Representation of a Kubernetes cluster that has been registered.
 */
 message KubernetesClusterSpec {
+
+    // Metadata for clusters provisioned from cloud providers.
+    message ProviderInfo {
+        oneof ProviderInfoType {
```

Comment:
nit: `ProviderInfoType` -> `provider_info_type`

also, can you add comments to the uncommented fields?